manifests/config.pp

```diff
@@ -44,8 +44,20 @@
 
   ensure_resource('file', $properties_dir, {'ensure' => 'directory', 'owner' => $user, 'group' => $group} )
 
+  #
+  # Checking if we need to deploy realm file
+  #  ugly, I know. Fix it if you know better way to do that
+  #
   if 'file' in $auth_types {
-
+    $_deploy_realm = true
+  }
+  if 'ldap_shared' in $auth_types {
+    $_deploy_realm = true
+  }
+  if 'active_directory_shared' in $auth_types {
+    $_deploy_realm = true
+  }
```

Comment:
that's exactly what i was asking, @rooty0.

i was just trying to think of a more puppety way of structuring that logic.